server/base/src/main/java/org/apache/accumulo/server/fs/PreferredVolumeChooser.java

```diff
@@ -39,73 +40,139 @@
+    if (env.hasTableId()) {
+      volumes = getPreferredVolumesForTable(env, localConf, options);
+      // localConf.getTableConfiguration(env.getTableId()).get(PREFERRED_VOLUMES_CUSTOM_KEY);
+    } else { // env.hasScope()
+      volumes = getPreferredVolumesForNonTable(env, localConf, options);
+      // localConf.getSystemConfiguration().get(PREFERRED_VOLUMES_SCOPED_KEY(env.getScope()));
     }
-    TableConfiguration tableConf = localConf.getTableConfiguration(env.getTableId());
-    final Map<String,String> props = new HashMap<>();
-    tableConf.getProperties(props, PREFERRED_VOLUMES_FILTER);
-    if (props.isEmpty()) {
-      log.warn("No preferred volumes specified. Defaulting to randomly choosing from instance volumes");
-      return super.choose(env, options);
+
+    // Randomly choose the volume from the preferred volumes
+    String choice = super.choose(env, volumes.toArray(EMPTY_STRING_ARRAY));
+    if (log.isTraceEnabled()) {
+      log.trace("Choice = " + choice);
     }
-    String volumes = props.get(PREFERRED_VOLUMES_CUSTOM_KEY);
+    return choice;
+  }
 
+  private List<String> getPreferredVolumesForTable(VolumeChooserEnvironment env, ServerConfigurationFactory localConf, String[] options)
+      throws AccumuloException {
     if (log.isTraceEnabled()) {
-      log.trace("In custom chooser");
-      log.trace("Volumes: " + volumes);
-      log.trace("TableID: " + env.getTableId());
+      log.trace("Looking up property " + TABLE_PREFERRED_VOLUMES + " for Table id: " + env.getTableId());
     }
-    // If the preferred volumes property was specified, split the returned string by the comma and add use it to filter the given options.
-    Set<String> preferred = parsedPreferredVolumes.get(volumes);
-    if (preferred == null) {
-      preferred = new HashSet<>(Arrays.asList(StringUtils.split(volumes, ',')));
-      parsedPreferredVolumes.put(volumes, preferred);
+    final TableConfiguration tableConf = localConf.getTableConfiguration(env.getTableId());
+    String volumes = tableConf.get(TABLE_PREFERRED_VOLUMES);
+
+    return getFilteredOptions(TABLE_PREFERRED_VOLUMES, volumes, options);
+  }
+
+  private List<String> getPreferredVolumesForNonTable(VolumeChooserEnvironment env, ServerConfigurationFactory localConf, String[] options)
+      throws AccumuloException {
+    String property = SCOPED_PREFERRED_VOLUMES(env.getScope());
+
+    if (log.isTraceEnabled()) {
+      log.trace("Looking up property: " + property);
+    }
+
+    AccumuloConfiguration systemConfiguration = localConf.getSystemConfiguration();
+    String volumes = systemConfiguration.get(property);
+
+    // only if the custom property is not set to we fallback to the default scoped preferred volumes
+    if (null == volumes) {
+      log.debug("Property not found: " + property + " using " + DEFAULT_SCOPED_PREFERRED_VOLUMES);
+      property = DEFAULT_SCOPED_PREFERRED_VOLUMES;
+      volumes = systemConfiguration.get(DEFAULT_SCOPED_PREFERRED_VOLUMES);
```

Comment:
Again, this one is handled differently, and will not fail as expected, if a scope's chooser is not set. And, falling back to a catch-all chooser for all scopes I'm not sure is a good idea.